dev-python/aiohttp/aiohttp-3.6.2.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="http client/server for asyncio"
+HOMEPAGE="https://pypi.org/project/aiohttp/"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 x86"
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+COMMON_DEPEND="
+	>=dev-python/async_timeout-3.0.0[${PYTHON_USEDEP}]
+	>=dev-python/attrs-17.3.0[${PYTHON_USEDEP}]
+	dev-python/chardet[${PYTHON_USEDEP}]
+	>=dev-python/multidict-4.5.0[${PYTHON_USEDEP}]
+	>=dev-python/yarl-1.0[${PYTHON_USEDEP}]
+	dev-python/idna-ssl[${PYTHON_USEDEP}]
+"
+DEPEND="
+	dev-python/cython[${PYTHON_USEDEP}]
+	test? (
+		${COMMON_DEPEND}
+		dev-python/async_generator[${PYTHON_USEDEP}]
+		dev-python/brotlipy[${PYTHON_USEDEP}]
+		>=dev-python/pytest-3.4.0[${PYTHON_USEDEP}]
+		dev-python/pytest-mock[${PYTHON_USEDEP}]
+		dev-python/pytest-timeout[${PYTHON_USEDEP}]
+		dev-python/trustme[${PYTHON_USEDEP}]
+		www-servers/gunicorn[${PYTHON_USEDEP}]
+	)
+"
+RDEPEND="${CDEPEND}"
+
+DOCS=( CHANGES.rst CONTRIBUTORS.txt README.rst )
+
+python_prepare_all() {
+	# FIXME
+	rm tests/test_pytest_plugin.py || die
+	sed -i -e 's:test_testcase_no_app:_&:' tests/test_test_utils.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+distutils_enable_sphinx docs \
+	'>=dev-python/alabaster-0.6.2' \
+	'dev-python/sphinxcontrib-asyncio' \
+	'dev-python/sphinxcontrib-blockdiag' \
+	'dev-python/sphinxcontrib-newsfeed' \
+	'dev-python/sphinxcontrib-spelling' \
+	'dev-python/sphinx' \
+	'dev-python/sphinx-aiohttp-theme'
+
+distutils_enable_tests pytest || die "Tests fail with ${EPYTHON}"
```

Comment:
`|| die` is not needed here.

I can't get the tests to run, 
```
>>> Test phase: dev-python/aiohttp-3.6.2
 * python3_6: running distutils-r1_run_phase python_test
================================== test session starts ==================================
platform linux -- Python 3.6.10, pytest-4.6.9, py-1.8.0, pluggy-0.13.1 -- /usr/bin/python3.6
cachedir: .pytest_cache
rootdir: /tmp/portage/dev-python/aiohttp-3.6.2/work/aiohttp-3.6.2
plugins: asyncio-0.11.0, timeout-1.3.4, mock-3.1.0
collecting ... collected 415 items / 46 errors / 369 selected

======================================== ERRORS =========================================
________________________ ERROR collecting aiohttp/test_utils.py _________________________
import file mismatch:
imported module 'aiohttp.test_utils' has this __file__ attribute:
  /tmp/portage/dev-python/aiohttp-3.6.2/work/aiohttp-3.6.2-python3_6/lib/aiohttp/test_utils.py
which is not the same as the test file we want to collect:
  /tmp/portage/dev-python/aiohttp-3.6.2/work/aiohttp-3.6.2/aiohttp/test_utils.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
_____________________ ERROR collecting tests/test_base_protocol.py ______________________
/usr/lib64/python3.6/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/usr/lib64/python3.6/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/usr/lib64/python3.6/site-packages/pluggy/manager.py:87: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/usr/lib64/python3.6/site-packages/_pytest/python.py:204: in pytest_pycollect_makeitem
    res = outcome.get_result()
/usr/lib64/python3.6/site-packages/pytest_asyncio/plugin.py:39: in pytest_pycollect_makeitem
    item = pytest.Function.from_parent(collector, name=name)
E   AttributeError: type object 'Function' has no attribute 'from_parent'
```
and this error repeats like 46 times. Any ideas? It also complained something about the file you modify with `sed` in this ebuild. Could it be related?

Reply:
The `AttributeError: type object 'Function' has no attribute 'from_parent'` means that you need to upgrade to at least pytest-5.4. A dependency should be added to the pytest-asyncio ebuild.

Reply:
I've fixed the pytest-asyncio dep in 93b9fca29cccb66846a5b95edc96fd05171c777a.

I've merged the aiohttp version bump in eb5c65238c8c617f2a7c3b88ca2af7e16a88fde0, and the aiofile ebuild in 7c34cc5a6165b8182ddc1614b1148f8b7fdd3554.